File: 0027-remove-element/0027-remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        slow = 0
        for fast in nums:
            if fast != val:
                nums[slow] = fast
                slow += 1
        return slow
        

File: 0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int fast = 0;
        int slow = 0;
        for (fast; fast < nums.size(); fast++) {
            if (nums[fast] != val) {
                nums[slow++] = nums[fast];
            }
        }
        return slow;
    }
};